### PR TITLE
Add Azkaban module with TensorFlowJob

### PR DESCRIPTION
**Pull request by hungj**

### CODE REVIEW

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        props.put(HADOOP_OPTS, String.format("%s %s", hadoopOps, hadoopGlobalOps));
+      } else {
+        props.put(HADOOP_OPTS, hadoopGlobalOps);
+      }
+    }
+  }
+
+  @Override
+  public String getJavaClass() {
+    return "com.linkedin.tony.TonyClient";
```
Comment (oliverhu) [inline tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: Switch to ClusterSubmitter

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  protected String getMainArguments() {
+    String args = super.getMainArguments();
+    info("All job props: " + getJobProps());
+    info("All sys props: " + getSysProps());
+    String srcDir = getJobProps().getString(TensorFlowJobArg.SRC_DIR.azPropName, "src");
+    args += " " + TensorFlowJobArg.SRC_DIR.tfParamName + " " + srcDir;
+
+    String hdfsClasspath = getJobProps().getString(TensorFlowJobArg.HDFS_CLASSPATH.azPropName, null);
+    if (hdfsClasspath != null) {
+      args += " " + TensorFlowJobArg.HDFS_CLASSPATH.tfParamName + " " + hdfsClasspath;
```
Comment (oliverhu) [inline tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: HDFS_CLASSPATH should be deprecated

File: build.gradle [hunk 3]
```diff
@@ -23,33 +23,43 @@ buildscript {
...
-        "sshd": "org.apache.sshd:sshd-core:1.1.0",
-        "testng": "org.testng:testng:6.4",
-        "text": "org.apache.commons:commons-text:1.4",
-        "zip4j": "net.lingala.zip4j:zip4j:1.3.2"
-    ]
+  azkaban: [
+    "az_core": "com.linkedin.azkaban:az-core:3.58.0",
+    "azkaban_common": "com.linkedin.azkaban:azkaban-common:3.58.0",
+    "azkaban_plugins": "com.linkedin.azkaban:az-hadoop-jobtype-plugin:3.58.0",
+    "azkaban_security": "com.linkedin.azkaban:azkaban-hadoop-security-plugin:3.58.0"
```
Comment (erwa) [inline build.gradle]: Can we use the full names `az_hadoop_jobtype_plugins` and `azkaban_hadoop_security_plugin` as our variable names? Azkaban has other types of plugins besides jobtype plugins, and also other types of security plugins.

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    List<String> classPath = super.getClassPaths();
+    classPath.add(tonyConfFile.getParent());
+    return classPath;
+  }
+
+  /**
+   * First refresh the extract
+   * once the status is found, log the results and cancel the job if
+   * the status was an error or a timeout
+   */
```
Comment (erwa) [inline tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: What is this comment talking about?

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java (hunk 2 above)
Comment (hungj) [reply tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: I don't think we should deprecate this. In general users should run with hdfs_classpath. I don't think it's a good practice to use the fatjar approach, even if it's easiest.

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java (hunk 1 above)
Comment (hungj) [reply tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: Same as below...ClusterSubmitter is useful for submitting from gateway quickly, but dependency management is much cleaner if hdfs_classpath is configured by users and uses the TonY thin jar.

File: build.gradle (hunk 3 above)
Comment (hungj) [reply build.gradle]: sure, fixed

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java (hunk 4 above)
Comment (hungj) [reply tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: No clue, removed

File: tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java (hunk 2 above)
Comment (oliverhu) [reply tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: What is the disadvantage of using a farjar here? it only makes user's life easier. 
- it avoids the jar conflicts problem
- it prevents the problem is a user forgot to run the cachedLib task

Comment (hungj) [reply tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java]: What jar conflict problem are you referring to? Using fatjar makes dependency conflicts worse. e.g if they are running one version of Hadoop which has a dependency with version X, and TonY is built against another version which pulls in that dependency with version Y, there will be a mismatch

**Pull request merged**